### PR TITLE
NEW TRAIT ROLE: Veteran Advisor

### PR DESCRIPTION
**Pull request by DrDiasyl**
## About The Pull Request
Adds a new Trait Job (jobs which only appear when the station gets a trait for it) - Veteran Advisor.
The weight of the trait is 2 which makes it very rare.

Veteran Security Advisor is a role designated as a (duh) Advisor for the HoS and Captain, as well as a mentor for all of the Officers. This is because it has an experience requirement of **100 SECURITY HOURS**, so no green Veterans.

Veteran Security Advisors are paraplegic (bonded to a wheelchair) and with PTSD which freezes their mood and sometimes causes hallucinations.
They get an M1911 pistol in a holster because that's badass.

<details>
<summary>THE IMAGES</summary>

ADVISOR
veteran advisor chair

THINGS HE SPAWNS WITH
veteran advisor shit

HIS UNIQUE MOODLET
veteran advisor desentized

SIGNUP BUTTON
signup button

</details>

## Why It's Good For The Game
More funny Trait Jobs to the game!
There is a certain lack of those, there is only Cargorilla and Bridge Assistant.

This job has big opportunities. Advisors can behave themselves like drill sergeants, or wise old mentors. Because they are kinda, as people only with 100 Security hours (yes those masochists exist) could play it when it appears. This will also open opportunities for retired Sec mains to tutor new generations without being directly involved in all of the stress.
I mostly took a look at the Goonstation "Nanotrasen Security Consultant" and Baystation "Senior Enlisted Advisor" as inspiration.

I tried to cut down any chances of this role being abused by power gamers for its M1911 pistol and Bridge access, due to this they are paraplegic so power gamers get disgusted, and if someone does somehow win Antagonists while in a wheelchair - props to them, pure skill.

**REMEMBER THAT THIS IS A TRAIT JOB!** This means they won't appear every round, but rather rarely! So they fit in as an occasional extra fun for the round.
## Changelog
:cl:
add: NEW TRAIT JOB: Veteran Security Advisor! Advise HoS and Captain on Security matters, mentor Security Officers. Note that they are paraplegic and are suffering PTSD due to their past experience.
/:cl:

### CODE REVIEW

**Comment by Wallemations** [comment]
Unique hallucinations for them where they hear gunshot and esword sounds at random to mimic their PTSD

**Comment by kawaiinick** [comment]
giving him telebaton along with a gun is fucked up though

**Comment by localbox48** [comment]
Tribunal Veteran LFWB

File: config/jobconfig.toml [hunk 1]
```diff
@@ -263,6 +263,13 @@
 "# Spawn Positions" = 5
 "# Total Positions" = 5
 
+[VETERAN_ADVISOR]
+"# Playtime Requirements" = 6000
```
Comment (Jacquerel) [inline config/jobconfig.toml:267]: I'm like 90% sure this does nothing on a station trait job.

**Review by Jacquerel — CHANGES_REQUESTED** [review]

**Comment by Jackal-boop** [comment]
are they a member of security? or are they a purely advisory role like lawyer

**Comment by Jacquerel** [comment]
I think having a role which shows up once in a blue moon and also has difficult job restrictions is a weird choice.
The button should not appear if you cannot sign up, and it kind of looks like it doesn't actually make any kind of played time check right now anyway.

**Comment by DrDiasyl** [comment]
> are they a member of security? or are they a purely advisory role like lawyer

They count as a Security member

**Comment by Admiral-Gigaton** [comment]
> I think having a role which shows up once in a blue moon and also has difficult job restrictions is a weird choice. The button should not appear if you cannot sign up, and it kind of looks like it doesn't actually make any kind of played time check right now anyway.

maybe a good few hours on HoS and/or cap instead?

**Comment by DrDiasyl** [comment]
> > I think having a role which shows up once in a blue moon and also has difficult job restrictions is a weird choice. The button should not appear if you cannot sign up, and it kind of looks like it doesn't actually make any kind of played time check right now anyway.
> 
> maybe a good few hours on HoS and/or cap instead?

100 hours is rather reasonable for a role that is supposed to be a Veteran, as most HoS mains have 100 hours.
Having too few hours will undermine the role.

**Comment by DrDiasyl** [comment]
> I think having a role which shows up once in a blue moon and also has difficult job restrictions is a weird choice. The button should not appear if you cannot sign up, and it kind of looks like it doesn't actually make any kind of played time check right now anyway.

@Jacquerel I think this line of code in the Trait Jobs already does this
```c
/datum/station_trait/job/can_display_lobby_button(client/player)
    var/datum/job/our_job = SSjob.GetJob(job_to_add::title)
    return our_job.player_old_enough(player) && ..()```

**Comment by DrDiasyl** [comment]
> Unique hallucinations for them where they hear gunshot and esword sounds at random to mimic their PTSD

Real now.

**Comment by DrDiasyl** [comment]
I start really believing that those check fails are unrelated

**Comment by jughu** [comment]
Make it 5k security hours it will be like the assistant cape but security :^)

**Comment by DrDiasyl** [comment]
> Make it 5k security hours it will be like the assistant cape but security :^)

Anyone who has 5k security hours is probably suffering PTSD irl

File: config/jobconfig.toml (hunk 1 above)
Comment (DrDiasyl) [reply config/jobconfig.toml:267]: This is resolved right, as this is in the trait job code. This shows the trait job sign up button only if person has fulfilled the time requirement.
```c
/datum/station_trait/job/can_display_lobby_button(client/player)
    var/datum/job/our_job = SSjob.GetJob(job_to_add::title)
    return our_job.player_old_enough(player) && ..()`

File: code/datums/brain_damage/special.dm [hunk 2]
```diff
@@ -426,3 +426,49 @@
...
+		/datum/hallucination/battle/gun/disabler,
+		/datum/hallucination/battle/gun/laser,
+		/datum/hallucination/battle/bomb,
+		/datum/hallucination/battle/e_sword,
+		/datum/hallucination/battle/harm_baton,
+		/datum/hallucination/battle/stun_prod,
+	)
+
+/datum/brain_trauma/special/ptsd/on_life(seconds_per_tick, times_fired)
+	if(owner.stat != CONSCIOUS || owner.IsSleeping() || owner.IsUnconscious())
```
Comment (Ghommie) [inline code/datums/brain_damage/special.dm:455]:
```suggestion
	if(owner.stat != CONSCIOUS)
```
The other two checks are redundant.

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Comment by DrDiasyl** [comment]
Okay, all of the new checks get stuck in queue as well

**Pull request merged**